.vscode/launch.json

```diff
@@ -0,0 +1,34 @@
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```

Comment:
Is it necessary to keep this?

Reply:
It's default text created by VSCode when a new launch.json file is created (I didn't write it).

Reply:
Well, I don't use vscode, I use JetBrains's

Reply:
I think we should not commit this file. The `.idea` folder is currently ignored, which serves the same purpose as `.vscode` but for JetBrains IntelliJ users.

This file may vary from user to user.

Reply:
@trusktr -- Standardizing debug configurations for IDEs is more akin to standardizing NPM and CI scripts than user preferences. Developers don't want to type `./node_modules/.bin/someBinary -a arg1 -b arg2 -c var3` for frequently run tasks, so we simplify by using `npm run something`. Similarly, I don't want to type  `npm run tests:jest -- -i /path/to/file.test.js -t "name of describe of test block"` just to run a single test, so running a debug script from the context of a file (so I don't have to type the path) and/or with a test name selected (so I don't have to type the test name) means I can just press a button to run a single test, making testing easier and more efficient. Win win.

There is no reason to omit these kinds of project-specific files from the repo because they should not vary between users. That's the whole purpose behind storing their configuration as files that can be managed by source control instead of storing them as user/app preferences. They should be managed like any other change to the project: users can add/modify them, submit changes as PRs, then they are reviewed and (hopefully) merged.

Reply:
Someone may not want to run in `integratedTerminal` for example. Maybe VS Code shouldn't mix personal configs with actually-useful/convenient project configs. Not sure why they do that. F.e. why should the software project dictate if the test runs in `integratedTerminal` or not?

But I do see the convenience (choose a file, highlight text with cursor, hit the button and it runs only that test). No harm in keeping it.

Reply:
We can ensure the project's required steps (f.e. `npm test`) are not in an editor-specific config (which is currently the case), and reserve the editor configs for conveniences only.

Reply:
@trusktr --

Agreed on not putting anything critical in editor-specific scripts/configs. These are convenience only. If folks using webstorm for example want to check in the `.idea` folder, that's fine with me as long as I won't be required to use webstorm to perform some critical function (and you don't do something silly like for an editor theme for everyone else).

As for integrated vs external terminals, this is where I think individuals are free to modify files in their local dev environment and tell git to ignore their own changes to those files. I went with the integrated terminal because it's the default selection and it seems the least intrusive, but folks can change it locally (or if everyone wants to use an external terminal, just file PR and we'll discuss).